Packs/CortexXDR/Playbooks/Cortex_XDR_-_Retrieve_File_v2.yml

```diff
@@ -0,0 +1,322 @@
+      {
+        "position": {
+          "x": 265,
+          "y": 50
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "1":
+    id: "1"
+    taskid: 7cb1327a-ad50-4f2d-8a36-27848573df25
+    type: regular
+    task:
+      id: 7cb1327a-ad50-4f2d-8a36-27848573df25
+      version: -1
+      name: Retrieve files
+      description: Retrieves files from selected endpoints. You can retrieve up to 20 files, from no more than 10 endpoints. At least one endpoint ID and one file path are necessary in order to run the command. After running this command, you can use the xdr-action-status-get command with returned action_id, to check the action status.
+      script: '|||xdr-file-retrieve'
+      type: regular
+      iscommand: true
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "7"
+    scriptarguments:
+      endpoint_ids:
+        complex:
+          root: inputs.endpoint_ids
+      generic_file_path:
+        complex:
+          root: inputs.file_path
+      interval_in_seconds:
+        simple: "60"
+      linux_file_paths:
+        complex:
+          root: inputs.linux_file_paths
+      mac_file_paths:
+        complex:
+          root: inputs.mac_file_paths
+      timeout_in_seconds:
+        simple: "600"
+      windows_file_paths:
+        complex:
+          root: inputs.windows_file_paths
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 265,
+          "y": 195
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "4":
+    id: "4"
+    taskid: a2712ae7-1100-4545-85a7-43770ff6fbd7
+    type: title
+    task:
+      id: a2712ae7-1100-4545-85a7-43770ff6fbd7
+      version: -1
+      name: Done
+      type: title
+      iscommand: false
+      brand: ""
+      description: ''
+    separatecontext: false
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": 265,
+          "y": 905
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "5":
+    id: "5"
+    taskid: 069bc12b-b96b-47ef-86dd-c000cdc518b6
+    type: condition
+    task:
+      id: 069bc12b-b96b-47ef-86dd-c000cdc518b6
+      version: -1
+      name: Retrieve file finished
+      description: Checks if the link to the file is ready.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "6"
+      "yes":
+      - "4"
+    separatecontext: false
+    conditions:
+    - label: "yes"
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              simple: File
```

Comment:
note that this is a global context key - which means that it can be nested through multiple sub-playbooks.
Hence, may I suggest checking if the XDR-retrieve command has a dedicated output for finishing?